tests/model_registry/rbac/conftest.py

```diff
@@ -250,3 +350,56 @@ def mr_access_role_binding(
         LOGGER.info(f"RoleBinding {binding.name} created successfully.")
         yield binding
         LOGGER.info(f"RoleBinding {binding.name} deletion initiated by context manager.")
+
+
+@pytest.fixture()
+def login_as_test_user(
+    api_server_url: str, original_user: str, test_idp_user_session: UserTestSession
+) -> Generator[None, None, None]:
+    LOGGER.info(f"Logging in as {test_idp_user_session.username}")
+    login_with_user_password(
+        api_address=api_server_url,
+        user=test_idp_user_session.username,
+        password=test_idp_user_session.password,
+    )
+    yield
+    LOGGER.info(f"Logging in as {original_user}")
+    login_with_user_password(
+        api_address=api_server_url,
+        user=original_user,
+    )
```

Comment:
why using logins instead of context? with context we should have easier code since we don't really need to keep track of anything other than the context name